### PR TITLE
iOS: Improve syntax for providing targeted pods

### PR DESCRIPTION
**Pull request by tido64**

### CODE REVIEW

**Comment by tido64** [comment]
@alloy: Would you say this is more idiomatic Ruby? Any better suggestions?

File: example/Podfile [hunk 1]
```diff
@@ -2,12 +2,12 @@ require_relative 'node_modules/react-native-test-app/test_app.rb'
...
-end
+add_app_pods = Proc.new {
+  pod 'Example-dev', :path => '.'
+}
+
+add_test_pods = Proc.new {
+  pod 'Example-Tests', :path => '.'
+}
+
+use_test_app!(__dir__, :add_app_pods => add_app_pods, :add_test_pods => add_test_pods)
```
Comment (alloy) [inline example/Podfile]: What do you think of something like the following?

```ruby
use_test_app!(__dir__) do |target|
  target.app do
    pod 'Example-dev', :path => '.'
  end

  target.test do
    pod 'Example-Tests', :path => '.'
  end
end
```

Comment (alloy) [reply example/Podfile]: I feel like `use_test_app` could be improved upon too, but don’t have a concrete suggestion right now. Will sleep on it.

Comment (tido64) [reply example/Podfile]: Definitely! Please feel free to make any ergonomic improvements.

I pushed some changes. Let me know if it's what you had in mind.

Comment (alloy) [reply example/Podfile]: Yup, looks great 👌

**Pull request merged**